frontend/src/app/shared/components/file-links/file-link-list/file-link-list.component.ts

```diff
@@ -158,33 +163,80 @@ export class FileLinkListComponent extends UntilDestroyedMixin implements OnInit
     }
   }
 
-  private setAuthenticationFailureState(fileLinkCount:number):void {
-    this.informationBoxHeader = this.text.infoBox.authenticationFailureHeader;
-    this.informationBoxContent = this.text.infoBox.authenticationFailureContent;
-    this.informationBoxButton = this.text.infoBox.loginButton;
-    this.informationBoxIcon = 'import';
+  private setAuthorizationFailureState(fileLinkCount:number):void {
+    this.prepareInformationBox(
+      this.text.infoBox.authorizationFailureHeader,
+      this.text.infoBox.authorizationFailureContent,
+      'import',
+    );
+
+    this.buttons.next([
+      new StorageActionButton(
+        this.text.infoBox.loginButton,
+        () => {
+          const nonce = uuidv4();
+          this.setAuthorizationCallbackCookie(nonce);
+          window.location.href = FileLinkListComponent.authorizationFailureActionUrl(
+            this.storage._links.authorize?.href as string,
```

Comment:
can we guard this accessor? if authorize is not available? we could add an early return before creating the nonce.